resources/building-mobile-extensions/dispatching-events-from-your-extension.md

```diff
@@ -115,84 +111,40 @@ public void loginButtonClicked() {
 
-You can have this code in a ViewController class or in one of your extensions public API classes:
+#### iOS
 
-```text
-// how to dispatch a paired event with an associated response callback
-NSError *eventError = nil;
-ACPExtensionEvent *event = [ACPExtensionEvent 
-extensionEventWithName:@"Dispatch Request Event with Callback"
-                  type:@"com.myCompany.eventType.custom"
-                source:@"com.myCompany.eventSource.request"
-                  data:@{}
-                 error:&eventError];
+```objective-c
+#import "ACPCore_iOS/ACPExtensionEvent.h"
 
-if (!event && eventError) {
-    NSLog(@"An error occurred while constructing event '%@': %ld", event.eventName, [eventError code]);
-    return;
-}
+- (IBAction) loginButtonClicked:(id)sender {
+    // construct the event to dispatch (see above)
+    ACPExtensionEvent* newEvent = ...;
 
-// dispatch the event and handle the callback
-NSError *dispatchError = nil;
-if ([ACPCore dispatchEventWithResponseCallback:event responseCallback:^(ACPExtensionEvent * _Nonnull responseEvent) {
-    NSLog(@"Response event received, with type %@, source %@ and data %@", responseEvent.eventType, responseEvent.eventSource, responseEvent.eventData);
-        // process event data
-} error:&dispatchError]) {
-    NSLog(@"Dispatched a paired request event '%@'", event.eventName);
-} else if (dispatchError) {
-    NSLog(@"An error occurred dispatching event '%@': %ld", event.eventName, [dispatchError code]);
+    // dispatch the event
+    NSError* error = nil;
+    if (![ACPCore dispatchEvent:newEvent error:&error] && error) {
+        NSLog(@"Error dispatching event %@:%ld", [error domain], [error code]);
+    };
 }
-...
 ```
 
-Register a listener for this event type and source in MyExtension.m:
+{% endtab %}
+{% endtabs %}
 
-```text
-// register a listener for a the request event type and source
-- (instancetype) init {
-    if (self = [super init]) {
-        NSError *error = nil;
-        if ([self.api registerListener:[MyExtensionListener class]
-                             eventType:@"com.myCompany.eventType.custom"
-                           eventSource:@"com.myCompany.eventSource.request"
-                                 error:&error]) {
-               NSLog(@"MyExtensionListener successfully registered for request events");
-         } else if (error) {
-            NSLog(@"There was an error registering MyExtensionListener: %ld", [error code]);
-         }
-...
-}
-```
+### Dispatch Paired Events
 
-Dispatch a response event when the request is received in the `hear` method of your Listener in MyListener.m:
+If you need to use a request `ACPExtensionEvent` \(iOS\) / `Event` \(Android\) as a trigger, and you have a callback to be called when the response paired event is sent, you can use the `dispatchEventWithResponseCallback` API from `ACPCore` \(iOS\) / `MobileCore` \(Android\). Then, the paired response event should be sent using the `dispatchResponseEvent` API.
```

Comment:
Please change the second sentence so it reads as follows:

"The paired response event is then sent by using the `dispatchResponseEvent` API."